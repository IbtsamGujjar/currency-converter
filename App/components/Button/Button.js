import React from 'react';
import PropTypes from 'prop-types';
import { TouchableOpacity, Image, StyleSheet, Text } from 'react-native';

import { Colors } from '../../../constants/colors';

const reverseImg = require('../../../assets/images/reverse.png');

const styles = StyleSheet.create({
  button: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    marginVertical: 20,
  },
  buttonIcon: {
    marginRight: 10,
    width: 20,
    height: 20,
  },
  buttonText: {
    fontSize: 16,
    color: Colors.white,
  },
});

const Button = ({ btnText, onPress, style }) => {
  return (
    <TouchableOpacity onPress={onPress} style={{ ...styles.button, ...style }}>
      <Image
        source={reverseImg}
        style={styles.buttonIcon}
        resizeMode="contain"
      />
      <Text style={styles.buttonText}>{btnText}</Text>
    </TouchableOpacity>
  );
};

Button.defaultProps = {
  btnText: 'Reverse Currencies',
  style: {},
  onPress: () => {},
};

Button.propTypes = {
  btnText: PropTypes.string,
  style: PropTypes.objectOf(PropTypes.any),
  onPress: PropTypes.func,
};

export { Button };
