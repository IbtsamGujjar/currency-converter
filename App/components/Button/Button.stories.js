import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';

import { Button } from './Button';
import { CenterView } from '../../../storybook/stories/decorator';
import { Colors } from '../../../constants/colors';

storiesOf('Button', module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add('Basic', () => (
    <Button
      onPress={action('tapped-default')}
      style={{ backgroundColor: Colors.grey }}
    />
  ));
