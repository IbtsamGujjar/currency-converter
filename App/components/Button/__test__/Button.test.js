import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from '../Button';

test('Button Snapshot', () => {
  const snapshot = renderer.create(<Button />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
