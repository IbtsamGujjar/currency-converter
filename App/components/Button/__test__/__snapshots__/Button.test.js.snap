// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "marginVertical": 20,
      "opacity": 1,
    }
  }
>
  <Image
    resizeMode="contain"
    source={1}
    style={
      Object {
        "height": 20,
        "marginRight": 10,
        "width": 20,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 16,
      }
    }
  >
    Reverse Currencies
  </Text>
</View>
`;
